Lib/test/test_generators.py

```
@@ -170,6 +170,46 @@ def f():
             g.send(0)
         self.assertEqual(next(g), 1)
 
+    def test_handle_frame_object_in_creation(self):
+
+        #Attempt to expose partially constructed frames
+        #See https://github.com/python/cpython/issues/94262
+
+        def cb(*args):
+            inspect.stack()
+
+        def gen():
+            yield 1
+
+        thresholds = gc.get_threshold()
+
+        gc.callbacks.append(cb)
+        gc.set_threshold(1, 0, 0)
+        try:
+            gen()
+        finally:
+            gc.set_threshold(*thresholds)
+            gc.callbacks.pop()
+
+        class Sneaky:
```

This test case passes even on main because the exception is caught early in try/except whereas the bug report does not catches the exception. Currently the exception is not propagated so the frame for generator is not created hence this test tests nothing.

---

I suggest to copy and paste the exact code reproducers from the bug report without any code changes and assert that `KeyboardInterrupt` is raised.

---

Exceptions in finalizers do not propagate. I've replaced the raise with `inspect.stack()` as that creates a frameobject for each frame on the stack.

---

> Exceptions in finalizers do not propagate. 

They do propagate but not in the usual sense that you can catch them, instead you have to use `sys.unraiseablehook` to check it. I was suggesting to use `test.support.catch_unraisable_exception` but `inspect.stack` is fine too.